source/common/quic/active_quic_listener.cc

```diff
@@ -211,6 +211,10 @@ uint32_t ActiveQuicListener::destination(const Network::UdpRecvData& data) const
   return connection_id_snippet % concurrency_;
 }
 
+size_t ActiveQuicListener::numReadsExpectedPerEventLoop() const {
+  return quic_dispatcher_.NumSessions();
```

Comment:
We can change the lower bound in readPacketsFromSocket() to something larger than 1. What's a reasonable number for that? Google internal code also use 1 as lower bound actually.

Reply:
I would say that for parity with TCP you probably need this to be 32 to 100 per active QUIC connection in the small number of streams case.  It's hard to say without loadtests.

Reply:
Please note that this number will *16 when using recvmmsg.  If the event loop duration is 500us and the lower bound is 100 reads, the expected max bandwidth is  100 * 16 * 1400 bytes / 0.0005s = 4.17GB/s = 33Gbps. Would a connection have such send rate?

Reply:
It would be good to make the packet read rate the same when recvmmsg is in-use / not-in-use

What are your thoughts in making these limits in number of packets and adjusting the number of calls in cases recvmmsg is used?

Regarding the math above, my concerned is about proxy behavior when event loop duration grows above 5ms, which would result in a decrease in the number of UDP packets that can be processed per second.  Also, what are the more common QUIC packets that you expect to receive?  I think they are ACKs and window updates which use smaller UDP packets; actual receive bps would be relatively small unless the proxy somehow is mostly receiving large POSTs from the clients.

Reply:
ACKs are most common packets if we are doing large download.

Reply:
I think that at least for downstreams downloads are more common than uploads.

Thoughts about making these limits configurable and accepting the same number of packets per wakeup for both recvmmsg and recvmsg ?